blockchain/blockchain.go

```diff
@@ -270,16 +272,53 @@ func (b *Blockchain) Receipt(hash *felt.Felt) (*core.TransactionReceipt, *felt.F
 
 func (b *Blockchain) L1Head() (*core.L1Head, error) {
 	var update *core.L1Head
-	if err := b.database.View(func(txn db.Transaction) error {
-		return txn.Get(db.L1Height.Key(), func(updateBytes []byte) error {
-			return encoder.Unmarshal(updateBytes, &update)
-		})
+
+	return update, b.database.View(func(txn db.Transaction) error {
+		var err error
+		update, err = l1Head(txn)
+		return err
+	})
+}
+
+func l1Head(txn db.Transaction) (*core.L1Head, error) {
+	var update *core.L1Head
+	if err := txn.Get(db.L1Height.Key(), func(updateBytes []byte) error {
+		return encoder.Unmarshal(updateBytes, &update)
 	}); err != nil {
 		return nil, err
 	}
 	return update, nil
 }
 
+func (b *Blockchain) IsL1Verified(blochHash *felt.Felt) bool {
```

Comment:
Should it be called bloc**k**Hash ? And also is it ok if caller receives `false` and makes assumption that block is not verified on L1 (but actually there is a storage failure or smth.)

Reply:
> Should it be called blockHash ?

I think so, what else can I call it?

> also is it ok if caller receives false and makes assumption that block is not verified on L1 (but actually there is a storage failure or smth.)

I see your point. I prefer to call `L1Head() (*core.L1Head, error)` and let the caller decide what to do with it (like I had before) then we can return the error for storage failure. I can write a private func in `handlers.go` to remove repetition, WDYT @omerfirmak?

I think any `Is*()` function, ideally, shouldn't return an error only a bool. However, I don't mind modifying function signature to:

```go
IsL1Verified(blochHash *felt.Felt) (bool, error)
```

Reply:
> I think so, what else can I call it?

but it's bloc**h**Hash now or you already renamed it?

> I prefer to call L1Head() (*core.L1Head, error) and let the caller decide what to do with it 

got it, makes sense

Reply:
> I can write a private func in handlers.go to remove repetition, WDYT @omerfirmak?

> I don't mind modifying function signature


Both works fine IMO, up to you.

Reply:
> but it's blochHash now or you already renamed it?

It was always blockHash you must be getting confused with something else

> Both works fine IMO, up to you.

Ok I am changing to how it was before

Reply:
Done